SCRD-5295 moved update processes out of InstallWizard

guglo opened the pull request:
This commit includes:
1. Moved update processes out of InstallWizard
2. Moved global states and persiste states to constants so
   they can be shared.
3. Moved some processes in InstallWizard into function so it can
   be overwritten.
4. Created UpdateWizard, as first step, it extends InstallWizard so
   core functions in InstallWizard can be shared.
5. Updated AddServer page and UpdateServer landing page to use UpdateWizard
6. Updated other components accordingly because of the move
7. Replaced componentWillReceiveProps with componentDidUpdate
8. Reduced unnecessary states.

GarySmith commented on src/InstallWizard.js:
```diff
@@ -20,40 +20,19 @@ import WizardProgress from './components/WizardProgress';
-const UPDATE_PERSISTED_STATES_VARS = [
-  'currentMenuName', 'processMenuName', 'processOperation', 'operationProps'
-];
+import {
```
I would prefer that these arrays not be put into `constants.js`, since they are very specific to the internal workings of this class and should not be exposed elsewhere. Furthermore, I don't even think they need to be constants since  they are each used only one line in the class (in the constructor).

guglo replied:
k, will look into it.

GarySmith commented on src/UpdateWizard.js:
```diff
@@ -0,0 +1,285 @@
+    };
+
+    // Indicate which of the above state variables are passed to wizard pages and can be set by them
+    this.globalStateVars =
```
Instead of having all of these global constants (which expose inner details of these classes too broadly), it would be better to do:
`this.globalStateVars = this.globalStateVars.concat(['currentMenuName, 'processMenuName' ...]);`

guglo replied:
k, will look into it.

GarySmith reviewed (changes requested):
This is a *great* improvement over the previous version. Thanks so much for doing this. Just one small change suggested here.

guglo commented:
need more testing.

Commit:
Rebased with latetest master and more updates

SCRD-5295 moved update processes out of InstallWizard

This commit includes:
1. Moved update processes out of InstallWizard
2. Moved global states and persiste states to constants so
   they can be shared.
3. Moved some processes in InstallWizard into function so it can
   be overwritten.
4. Created UpdateWizard, as first step, it extends InstallWizard so
   core functions in InstallWizard can be shared.
5. Updated AddServer page and UpdateServer landing page to use UpdateWizard
6. Updated other components accordingly because of the move
7. Replaced componentWillReceiveProps with componentDidUpdate
8. Reduced uncessary states.

Commit:
Removed un-necessary updates

Commit:
minor cleanup

GarySmith reviewed (approved):
LGTM. Lots of good cleanup here.
